### PR TITLE
Output & error handling

### PR DESCRIPTION
**Pull request by tommydeboer**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/system tested
- [ ] User documentation updated
- [ ] Clean commits
- [ ] Updated CHANGELOG.md

### CODE REVIEW

**Comment by tommydeboer** [comment]
[[#7970] Error handling & output](https://trello.com/c/emwEeaA9/225-7970-error-handling-output)

File: src/molgenis/bbmri_eric/_cli.py [hunk 1]
```diff
@@ -67,18 +67,20 @@ def _get_username_password(args) -> Tuple[str, str]:
...
+            nodes = national_nodes.get_external_nodes(args.nodes)
+        eric.stage_external_nodes(nodes)
     elif args.action == "publish":
         if all_nodes:
-            eric.publish_all_nodes()
+            nodes = national_nodes.get_all_nodes()
         else:
-            eric.publish_nodes(nnodes.get_nodes(args.nodes))
+            nodes = national_nodes.get_nodes(args.nodes)
+        eric.publish_nodes(nodes)
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/_cli.py:84]: Missing else

File: src/molgenis/bbmri_eric/_stager.py [hunk 2]
```diff
@@ -1,48 +1,56 @@
...
 from molgenis.bbmri_eric._model import ExternalServerNode, TableType
 from molgenis.bbmri_eric.bbmri_client import BbmriSession
+from molgenis.bbmri_eric.errors import EricError
+from molgenis.bbmri_eric.printer import Printer
 from molgenis.client import MolgenisRequestError
 
 
 class Stager:
-    def __init__(self, session: BbmriSession):
+    def __init__(self, session: BbmriSession, printer: Printer = None):
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/_stager.py:9]: yagni?

File: src/molgenis/bbmri_eric/eric.py [hunk 3]
```diff
@@ -21,36 +24,101 @@ def __init__(self, session: BbmriSession):
...
         directory.
 
         Parameters:
             nodes (List[Node]): The list of nodes to publish
         """
-        Publisher(self.session).publish(nodes)
+        report = ErrorReport(nodes)
+        publisher = Publisher(self.session, self.printer)
+        for node in nodes:
+            self.printer.reset()
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/eric.py:62]: what does it reset?

File: src/molgenis/bbmri_eric/printer.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    def __init__(self):
+        self.indents = 0
+
+    def indent(self):
+        self.indents += 1
+
+    def dedent(self):
+        self.indents = max(0, self.indents - 1)
+
+    def reset(self):
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/printer.py:17]: resetIndent ?

**Review by jelmerveen — CHANGES_REQUESTED** [review]

File: src/molgenis/bbmri_eric/_cli.py (hunk 1 above)
Comment (jelmerveen) [reply src/molgenis/bbmri_eric/_cli.py:86]: Still missing?

**Review by jelmerveen — APPROVED** [review]

**Pull request merged**